plugins/scanners/scanoss/src/main/kotlin/ScanOssResultParser.kt

```diff
@@ -135,34 +147,37 @@ private fun getCopyrightFindings(details: ScanFileDetails): List<CopyrightFindin
 
 /**
  * Get the snippet findings from the given [details]. If a snippet returned by SCANOSS contains several Purls,
- * several snippets are created in ORT each containing a single Purl.
+ * the function uses the first PURL as the primary identifier while storing all PURLs in additionalData
+ * to preserve the complete information.
  */
-private fun getSnippets(details: ScanFileDetails): Set<Snippet> {
+private fun getSnippets(details: ScanFileDetails): List<Snippet> {
     val matched = requireNotNull(details.matched)
-    val fileUrl = requireNotNull(details.fileUrl)
+    val ossFile = requireNotNull(details.file)
     val ossLines = requireNotNull(details.ossLines)
     val url = requireNotNull(details.url)
     val purls = requireNotNull(details.purls)
 
     val license = getUniqueLicenseExpression(details.licenseDetails.toList())
 
     val score = matched.substringBeforeLast("%").toFloat()
-    val locations = convertLines(fileUrl, ossLines)
+    val ossLocations = convertLines(ossFile, ossLines)
     // TODO: No resolved revision is available. Should a ArtifactProvenance be created instead ?
     val vcsInfo = VcsHost.parseUrl(url.takeUnless { it == "none" }.orEmpty())
     val provenance = RepositoryProvenance(vcsInfo, ".")
 
-    return buildSet {
-        purls.forEach { purl ->
-            locations.forEach { snippetLocation ->
-                add(Snippet(score, snippetLocation, provenance, purl, license))
-            }
-        }
+    // Store all PURLs in additionalData to preserve the complete information.
+    val additionalData = mapOf(
+        "all_purls" to purls.joinToString(", ")
```

Comment:
How about instead using the purls as keys, and use empty values for `additionalData`? That way purls can be retrieved back easier than parsing a value string.

Reply:
Also, the first purl IMO should be omitted here, as that one is already used as the primary purl for the snippet below.

Reply:
Maybe an approach like

```kotlin
val purls = requireNotNull(details.purls).toMutableList()
...
val primaryPurl = purls.removeFirstOrNull().orEmpty()
```

works, and afterwards put the remaining `purls` into `additionalData`.